src/dash-adapter.js

```diff
@@ -906,7 +910,26 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
       targetBufferVal = this._videoElement.duration - this._videoElement.currentTime;
     }
 
-    targetBufferVal = Math.min(targetBufferVal, this._shaka.getConfiguration().streaming.bufferingGoal);
+    targetBufferVal = Math.min(targetBufferVal, this._shaka.getConfiguration().streaming.bufferingGoal + this._getCurrentSegmentLength());
     return targetBufferVal;
   }
+
+  _getCurrentSegmentLength(): number {
+    const activeTrack = this._getActiveTrack();
+    const activeTrackId = activeTrack ? activeTrack.id : NaN;
+    let segmentLength = 0;
+    if (!isNaN(activeTrackId)) {
+      for (let i = 0; i < this._shaka.getManifest().periods.length; i++) {
```

Comment:
you can store `const periods = this._shaka.getManifest().periods;` and it will make the rest of the access a lot less verbose.